### PR TITLE
Fixes build errors on gcc < 5.2 and mcst compilers

### PR DESCRIPTION
**Pull request by Arenoros**

### CODE REVIEW

File: build/Jamfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ import ../../config/checks/config : requires ;
 
 project boost/charconv ;
 
-local SOURCES = from_chars.cpp to_chars.cpp ;
+local SOURCES = from_chars.cpp to_chars.cpp dragonbox.cpp;
```
Comment (pdimov) [inline build/Jamfile]: You need a space before the semicolon here. The jam/b2 language requires whitespace between tokens.

**Comment by mborland** [comment]
How much older than GCC 5.2 are you trying to make this work for? You can add the following lines to `.drone.jsonnnet` to cover the GCC 4.X versions:

```
linux_pipeline(
        "Linux 14.04 GCC 4.8* 32/64",
        "cppalliance/droneubuntu1404:1",
        { TOOLSET: 'gcc', COMPILER: 'g++', CXXSTD: '03,11', ADDRMD: '32,64' },
    ),

    linux_pipeline(
        "Linux 14.04 GCC 4.9 32/64",
        "cppalliance/droneubuntu1404:1",
        { TOOLSET: 'gcc', COMPILER: 'g++-4.9', CXXSTD: '03,11', ADDRMD: '32,64' },
        "g++-4.9-multilib",
        [ "ppa:ubuntu-toolchain-r/test" ],
    ),
```

For historical context support for GCC 4.X were dropped specifically for this issue.

**Comment by pdimov** [comment]
That's not going to work because b2 no longer supports building under 14.04. It should be something like
```
linux_pipeline(
        "Linux 16.04 GCC 4.8 32/64",
        "cppalliance/droneubuntu1604:1",
        { TOOLSET: 'gcc', COMPILER: 'g++-4.8', CXXSTD: '03,11', ADDRMD: '32,64' },
        "g++-4.8-multilib",
    ),

    linux_pipeline(
        "Linux 16.04 GCC 4.9 32/64",
        "cppalliance/droneubuntu1604:1",
        { TOOLSET: 'gcc', COMPILER: 'g++-4.9', CXXSTD: '03,11', ADDRMD: '32,64' },
        "g++-4.9-multilib",
    ),
```

**Comment by Arenoros** [comment]
cross build for armv7-linux-gcc 4.8.2

**Comment by pdimov** [comment]
There seems to be something wrong with msvc-14.0 (VS 2015), though.

**Comment by Arenoros** [comment]
the error is due to the warning that dragonbox.cpp does not create obj files because it should not create anything for msvc, in 4.9/4.8 the problem is with the conversion warning, I don't know how relevant they are at all. Is it possible for individual targets to ignore these warnings or is there a way to fix them?

**Comment by pdimov** [comment]
```
compile-c-c++ bin.v2\libs\charconv\build\msvc-14.0\debug\address-model-32\cxxstd-14-iso\link-static\threading-multi\dragonbox.obj
dragonbox.cpp
libs\charconv\build\..\src\dragonbox.cpp(9): error C2373: 'cache': redefinition; different type modifiers
D:\a\charconv\boost-root\boost/charconv/detail/dragonbox/dragonbox_common.hpp(771): note: see declaration of 'cache'
libs\charconv\build\..\src\dragonbox.cpp(14): error C2373: 'cache': redefinition; different type modifiers
D:\a\charconv\boost-root\boost/charconv/detail/dragonbox/dragonbox.hpp(860): note: see declaration of 'cache'
```
This is an error, not a warning.

The -Wconversion warnings are unrelated to this PR. Since we didn't have 4.8/4.9 in CI, we haven't noticed those. Matt may be able to do something about them if he decides to support 4.8/4.9.

**Comment by mborland** [comment]
Since you seem to be having similar build problems as I did I am going to close this. Frankly I am not interested in adding the maintenance burden of GCC 4.X to the library, and I do not want you to waste your time.

**Comment by Arenoros** [comment]
let me finalize my theories, because it is not realistic for me to deploy all tests locally, I just want to achieve that all tests pass successfully, and the further decision to include or not is not very important to me.

**Pull request closed without merging**